Source/SpatialGDK/Private/SpatialActorChannel.cpp

```diff
@@ -366,42 +367,44 @@ bool USpatialActorChannel::ReplicateActor()
 	{		
 		if (RepFlags.bNetInitial && bCreatingNewEntity)
 		{
-			// When a player is connected, a FUniqueNetIdRepl is created with the players worker ID. This eventually gets stored
-			// inside APlayerState::UniqueId when UWorld::SpawnPlayActor is called. If this actor channel is managing a pawn or a 
-			// player controller, get the player state.
-			FString PlayerWorkerId;
-			APlayerState* PlayerState = Cast<APlayerState>(Actor);
-			if (!PlayerState)
+			if (!Actor->IsFullNameStableForNetworking() || SpatialNetDriver->GetSpatialInterop()->CanSpawnReplicatedStablyNamedActors())
```

Comment:
How does this handle the race condition where this server will be able to replicated stably named actors, but hasn't been given authority yet over the gsm? Should we ensure that bCreatingNewEntity stays true?

Reply:
This block of code should only be reached after reserve entity id requests are sent out, which does queueing logic to prevent race conditions.
However, I will dig into this, since I am not completely confident.

Reply:
Dug into it, yes, that block cannot be reached before the reserve entity id response has been received:
At the top of ReplicateActor():
```
if (!IsReadyForReplication())
{
	return false;
}
```
and:
```
FORCEINLINE bool IsReadyForReplication() const
{
	// Wait until we've reserved an entity ID.		
	return ActorEntityId != FEntityId{};
}
```

Reply:
I can add a comment

Reply:
Thanks for the context, I don't think a comment is necessary.